libgadget/cooling.c

```diff
@@ -1034,31 +1042,21 @@ static void InitUVF(void) {
         interp_init_dim(&UVF.Finterp, 0, UVF.Zbins[0], UVF.Zbins[UVF.N_Zbins - 1]);
     }
 
-    UVF.Nside = size;
-
-    int Nside = UVF.Nside;
-    /* This is kinda big, so we move it to mymalloc (leaving more free space for
-     * system /MPI */
-    UVF.Table = mymalloc("Zreion", (sizeof(double) * Nside) * (Nside * Nside));
```

Comment:
Size is not initialized here! And is copied into Nside. We could also fix this in fewer changed lines by just moving the mymalloc after h5readdouble

Reply:
I see. Basically the entire Zerion table can be corrupt depending on the random number stored in size.
This explained why I was seeing ridiculously large memory usage from Zerion in BlueTides.